In [108]:
from __future__ import print_function
__author__ = "Sung Hoon Yang, CUSP NYU 2018"
import numpy as np
import pandas as pd
import matplotlib
font = {'family' : 'normal',
        'weight' : 'bold',
        'size'   : 88}

matplotlib.rc('font', **font)
import matplotlib.pyplot as plt
import seaborn as sns
import os
import re
%matplotlib inline
np.random.seed(999)

"""
Set up PUIDATA env var
"""
import os
os.environ["PUIDATA"] = "%s/fall18/PUI/PUIDATA"% os.getenv("HOME")

## Assignment 3: Finish z-test lab and turn it in as a notebook .

I am looking for here is: seeing a good Null/alternative hypothesis statement and treatment, with a clear Null and Alternative spelled out AND written out as a formula, and a good interpretation of the Z value you obtain in terms of ability or inability to reject the Null Hypothesis. 
Here is the forumla

<img src="http://bit.ly/2N3HGT6" align="center" border="0" alt="Z = \frac{\mu_{pop} - \mu_{sample}}{\sigma / \sqrt{N}}" width="154" height="44"/>

This is also in the slides attached (in a more readable format).

The chapter of _Statistics In a Nutshell_ that covers these topics is called Inferential statistics. 
It is chapter 3 in the hard copies of the book in the CUSP library, 
but it was moved to chapter 7 in the online book version which is in the link. Same content more or less.


### GRADING: 

Your notebook must display
- the complete formulation of the hypothesis (Null and Alternative) to be tested in words and formula
- the download of the data (which is in https://github.com/fedhere/PUI2018_fb55/blob/master/Lab4_fb55/times.txt, but you must get the raw data!)
- the calculation of the z statistics (with the given formula and the data processed from the data file)
- the comparison of the statistis with the significance threshold and the conclusions about the Null Hypothesis



In [109]:
!cd /nfshome/shy256/fall18/PUI/PUIDATA; curl -O https://raw.githubusercontent.com/fedhere/PUI2018_fb55/master/Lab4_fb55/times.txt

  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  1179  100  1179    0     0  10497      0 --:--:-- --:--:-- --:--:-- 10526


In [110]:
df = pd.read_csv(os.path.join(os.getenv('PUIDATA'), 'times.txt'), names=['val'])
df.head(7)

val
0  31.622239
1  32.821376
2  30.229101
3  31.413766
4  39.010550
5  34.822079
6  39.871884

In [111]:
__POPULATION_MEAN__ = 36
__POPULATION_STD__ = 6 

#### Hypothesis Testing
I design my Hypothesis Testing as such:
$$
H_0: T_{new} - T_{old} \ge 0
$$
$$
H_1: T_{new} - T_{old} \lt 0
$$
Since our alternative hypothesis captures inequality, the test will be two-tailed. I will perform the testing with 
$$
\alpha \in \{0.1, 0.05, 0.01\}
$$
    

In [112]:
from math import sqrt
_ex, _sd, N = df.mean().val, df.std().val, len(values)
df = pd.DataFrame(values, columns=['val'])
df['z-stat'] = (df['val'] - __POPULATION_MEAN__) / (__POPULATION_STD__ / sqrt(N))

In [113]:
# Calculating Confidence Intervals for one-tailed test.
p_vals = {
    .1: 1.282,
    .05: 1.645,
    .01: 1.96
}
print("z stat for %.4f: %.2f" % (_ex, (_ex - __POPULATION_MEAN__) / (__POPULATION_STD__ / sqrt(N))))
for a, p in p_vals.items():
    print("alpha: %.2f\t| z-score: %.2f\t" % (a, p))
    print("This is %d%% Confidence Interval (C.I.)" % (100 * a))
    print("C.I: (%.2f, inf)" % (-1.0 * p * __POPULATION_STD__ / sqrt(N) + __POPULATION_MEAN__))


z stat for 34.4662: -2.33
alpha: 0.10	| z-score: 1.28	
This is 10% Confidence Interval (C.I.)
C.I: (35.16, inf)
alpha: 0.05	| z-score: 1.65	
This is 5% Confidence Interval (C.I.)
C.I: (34.92, inf)
alpha: 0.01	| z-score: 1.96	
This is 1% Confidence Interval (C.I.)
C.I: (34.71, inf)


#### According to Confidence Intervals
There is 
* 10 % chance that population mean is outside (35.16, inf). This does not include 34.2689. Z-stat for 34.2689: -2.63 is also smaller than -1.28. So the test result is statistically significant and we reject Null Hypothesis.
* 5 % chance that population mean is outside (34.92, inf). This does not include 34.2689. Z-stat for 34.2689: -2.63 is also smaller than -1.65. So the test result is statistically significant and we reject Null Hypothesis.
* 1 % chance that population mean is outside (34.71, inf). This does not include 34.2689. Z-stat for 34.2689: -2.63 is also smaller than -1.96. So the test result is statistically significant and we reject Null Hypothesis.



In [114]:
## End of Notebook